File: src/com/Torvald/Terrarum/GameControl/KeyMap.java
package com.Torvald.Terrarum.GameControl;

import java.util.Hashtable;

/**
 * Created by minjaesong on 15-12-31.
 */
public class KeyMap {

    public static Hashtable<EnumKeyFunc, Integer> map_code = new Hashtable<>();

    public static void build(){

        map_code.put(EnumKeyFunc.MOVE_UP, Key.E);
        map_code.put(EnumKeyFunc.MOVE_LEFT, Key.S);
        map_code.put(EnumKeyFunc.MOVE_DOWN, Key.D);
        map_code.put(EnumKeyFunc.MOVE_RIGHT, Key.F);
        map_code.put(EnumKeyFunc.JUMP, Key.SPACE);
        map_code.put(EnumKeyFunc.UI_CONSOLE, Key.GRAVE);
        map_code.put(EnumKeyFunc.UI_BASIC_INFO, Key.F3);
    }

    public static int getKeyCode(EnumKeyFunc fn) {
        return map_code.get(fn);
    }

    public static void set(EnumKeyFunc func, int key){
        map_code.put(func, key);
    }

}

File: src/net/torvald/terrarum/gamecontroller/KeyMap.kt
package net.torvald.terrarum.gamecontroller

import net.torvald.terrarum.Terrarum
import java.util.Hashtable

/**
 * Created by minjaesong on 15-12-31.
 */
object KeyMap {

    var map_code = Hashtable<EnumKeyFunc, Int>()

    init {
        map_code.put(EnumKeyFunc.MOVE_UP, Terrarum.getConfigInt("keyup"))
        map_code.put(EnumKeyFunc.MOVE_LEFT, Terrarum.getConfigInt("keyleft"))
        map_code.put(EnumKeyFunc.MOVE_DOWN, Terrarum.getConfigInt("keydown"))
        map_code.put(EnumKeyFunc.MOVE_RIGHT, Terrarum.getConfigInt("keyright"))
        map_code.put(EnumKeyFunc.JUMP, Terrarum.getConfigInt("keyjump"))
        map_code.put(EnumKeyFunc.UI_CONSOLE, Key.GRAVE)
        map_code.put(EnumKeyFunc.UI_BASIC_INFO, Key.F3)
    }

    fun getKeyCode(fn: EnumKeyFunc): Int {
        return map_code[fn]!!
    }

    fun set(func: EnumKeyFunc, key: Int) {
        map_code.put(func, key)
    }

}
